packages/devextreme/js/__internal/scheduler/appointment_popup/appointment_popup.test.ts

```diff
@@ -422,51 +416,387 @@ describe('Appointment Popup Form', () => {
 
       scheduler.showAppointmentPopup(firstAppointment);
-      POM.popup.getSwitchByName('repeat').click();
 
-      POM.popup.selectRadio('After');
+      // @ts-expect-error
+      const repeatEditor = $(POM.popup.repeatEditor).dxSelectBox('instance');
+      repeatEditor.option('value', 'weekly');
 
-      POM.popup.getSaveButton().click();
+      scheduler.hideAppointmentPopup();
 
       scheduler.showAppointmentPopup(secondAppointment);
-      POM.popup.getSwitchByName('repeat').click();
 
-      const radioValue = POM.popup.getSelectedRadioValue();
+      // @ts-expect-error
+      const repeatEditorSecond = $(POM.popup.repeatEditor).dxSelectBox('instance');
+      const value = repeatEditorSecond.option('value');
+
+      expect(value).toBe('never');
+    });
+
+    it('Recurrence editor container should be visible after changing its visibility value', async () => {
+      const appointment = {
+        text: 'Test Appointment',
+        startDate: new Date(2017, 4, 1, 9, 30),
+        endDate: new Date(2017, 4, 1, 11),
+      };
+
+      const { POM, scheduler } = await createScheduler(getDefaultConfig());
+
+      scheduler.showAppointmentPopup(appointment);
+
+      const recurrenceGroup = $(POM.popup.element).find('.dx-scheduler-form-recurrence-group');
+
+      expect(recurrenceGroup.hasClass('dx-scheduler-form-recurrence-hidden')).toBe(true);
+
+      // @ts-expect-error
+      const repeatEditor = $(POM.popup.repeatEditor).dxSelectBox('instance');
+      repeatEditor.option('value', 'weekly');
+
+      expect(recurrenceGroup.hasClass('dx-scheduler-form-recurrence-hidden')).toBe(false);
+    });
+
+    it('Should discard changes when canceling from recurrence form', async () => {
+      const data = [
+        {
+          text: 'meet',
+          startDate: new Date(2017, 4, 22, 9, 30),
+          endDate: new Date(2017, 4, 22, 11, 30),
+        },
+      ];
+
+      const { POM, scheduler } = await createScheduler({
+        dataSource: data,
+        currentDate: new Date(2017, 4, 22),
+      });
+
+      POM.openPopupByDblClick('meet');
+
+      // @ts-expect-error
+      const repeatEditor = $(POM.popup.repeatEditor).dxSelectBox('instance');
+      repeatEditor.option('value', 'weekly');
+
+      scheduler.hideAppointmentPopup(true);
+
+      POM.openPopupByDblClick('meet');
+
+      POM.popup.getEditSeriesButton().click();
+
+      // @ts-expect-error
+      const repeatEditor2 = $(POM.popup.repeatEditor).dxSelectBox('instance');
+      repeatEditor2.option('value', 'daily');
+
+      POM.popup.getBackButton().click();
+
+      scheduler.hideAppointmentPopup();
+
+      POM.openPopupByDblClick('meet');
+
+      POM.popup.getEditSeriesButton().click();
+
+      // @ts-expect-error
+      const repeatEditor3 = $(POM.popup.repeatEditor).dxSelectBox('instance');
+      expect(repeatEditor3.option('value')).toBe('weekly');
+    });
+
+    describe('Recurrence frequency types', () => {
```

Comment:
Essentially this test group is a collection of markup tests that check presence of elements on the page. Taking into account that we have visual test for each of the frequencies, do we need these tests?

If visual tests already do these tests' job, then I suggest to remove them